drink/src/session.rs

```diff
@@ -93,33 +124,58 @@ impl Session {
             self.gas_limit,
         );
 
-        match &result.result {
+        let ret = match &result.result {
             Ok(exec_result) if exec_result.result.did_revert() => {
                 Err(SessionError::DeploymentReverted)
             }
             Ok(exec_result) => {
-                self.deploy_returns.push(exec_result.account_id.clone());
-                self.deploy_results.push(result);
-                Ok(self)
+                let address = exec_result.account_id.clone();
+                self.deploy_returns.push(address.clone());
+                Ok(address)
             }
             Err(err) => Err(SessionError::DeploymentFailed(*err)),
-        }
+        };
+
+        self.deploy_results.push(result);
+        ret
     }
 
-    pub fn call(mut self, message: &str, args: &[&str]) -> Result<Self, SessionError> {
+    pub fn call_and(
+        mut self,
+        address: Option<AccountId32>,
+        message: &str,
+        args: &[String],
+    ) -> Result<Self, SessionError> {
+        self.call(address, message, args).map(|_| self)
+    }
+
+    pub fn call(
```

Comment:
This is not very important but actually for a public API I'd prefer two methods:
```rust
pub fn call(&mut self, message: &str, args: &[String])
```
and:
```rust
pub fn call_with_address(&mut self, address: AccountId32, message: &str, agrs: &[String])
```
And have them both call a non-public `call_internal` method with a signature just like `call` now.
Just thinking out loud :)

Reply:
done :ok_hand: